graph/client/webpack.config.js

```diff
@@ -1,20 +1,157 @@
+const { join } = require('node:path');
 // nx-ignore-next-line
-const { withNx, composePlugins } = require('@nx/webpack');
+const { NxWebpackPlugin } = require('@nx/webpack');
 // nx-ignore-next-line
-const { withReact } = require('@nx/react');
+const { NxReactWebpackPlugin } = require('@nx/react');
 
-module.exports = composePlugins(withNx(), withReact(), (config, context) => {
-  return {
-    ...config,
-    resolve: {
-      ...config.resolve,
-      alias: {
-        ...config.resolve.alias,
-        react: 'preact/compat',
-        'react-dom/test-utils': 'preact/test-utils',
-        'react-dom': 'preact/compat', // Must be below test-utils
-        'react/jsx-runtime': 'preact/jsx-runtime',
-      },
+module.exports = {
+  output: {
+    path: join(__dirname, '../../build/apps/graph'),
+  },
+  devServer: {
+    port: getPort(process.env.NX_TASK_TARGET_CONFIGURATION),
+    historyApiFallback: {
+      disableDotRule: true,
     },
-  };
-});
+  },
+  resolve: {
+    alias: {
+      react: 'preact/compat',
+      'react-dom/test-utils': 'preact/test-utils',
+      'react-dom': 'preact/compat', // Must be below test-utils
+      'react/jsx-runtime': 'preact/jsx-runtime',
+    },
+  },
+  plugins: [
+    new NxWebpackPlugin({
+      index: './src/index.html',
+      compiler: 'babel',
+      main: './src/main.tsx',
+      tsConfig: './tsconfig.app.json',
+      styles: ['./src/styles.css'],
+      scripts: [],
+      assets: getAssets(process.env.NX_TASK_TARGET_CONFIGURATION),
```

Comment:
This depends on being run within Nx? :thinking: Is there a more native webpack way to specify a configuration or arbitrary variable was used?

Reply:
Not really. Only the mode is available, but it is strictly for `development`, `production`, `none`. The other option is to use environment variables, but it is clunky.

I think we should re-examine how `graph-client` and `e2e-graph-client` is setup. Emily can take that one as a future improvement.